src/ql/arch/cc/resources/hwconf_default.json

```diff
@@ -248,682 +367,1323 @@
             "duration": 20,
             "cc": {
                 "ref_signal": "single-qubit-mw",
-                "static_codeword_override": [0]
+                "static_codeword_override": [30]
             }
         },
         "cw_31": {
-            "prototype": ["U:qubit"],
+            "prototype": ["X:qubit"],
             "duration": 20,
             "cc": {
                 "ref_signal": "single-qubit-mw",
-                "static_codeword_override": [1]
+                "static_codeword_override": [31]
             }
         },
-        "fl_cw_00": {
-            "prototype": ["U:qubit", "U:qubit"],
-            "duration": 80,
+
+        //**************************************************************************************************************
+        // measure
+        //**************************************************************************************************************
+
+        // allow decompositions that prepend measurement with microwave gate
+        "_do_measure": {
+            "prototype": ["M:qubit"],
+            "duration": 800,
             "cc": {
-                "ref_signal": "two-qubit-flux",
-                "static_codeword_override": [0]
+                "signal": [
+                    {   "type": "measure",
+                        "operand_idx": 0,
+                        "value": ["dummy"]          // Future extension: specify output and weight, and generate code word
+                    }
+                ],
+                "static_codeword_override": [0]   // FIXME
             }
         },
-        "fl_cw_01": {
-            "prototype": ["U:qubit", "U:qubit"],
-            "duration": 80,
+        "measure": {
+            "prototype": ["M:qubit"],
+            "duration": 800,
+            "decomposition": {
+                "into": "_do_measure op(0)"
+            }
+        },
+        // FIXME: experiment to prefix some measurements with rx12
+//        "measure q6": {
+//            "prototype": ["M:qubit"],
+//            "duration": 820,
+//            "decomposition": {
+//                "into": [
+//                    "rx12 q[6];",
+//                    "_do_measure q[6]"
+//                ]
+//            }
+//        },
+
+        //**************************************************************************************************************
+        // flux
+        //**************************************************************************************************************
+
+        // helpers for CZ gate decomposition, flux_lutman compatible
+        //
+        // NB: the '*_park' instructions now use prototype "I:qubit" for the park parameter, which makes the scheduler
+        // *ignore* that parameter. This allows for parallel scheduling of multiple instructions that perform the same
+        // park, but can also arise in conflicts (signal conflict in the CC backend) if a qubit is both used for parking
+        // and another fluxing operation.
+
```

Comment:
There used to be a resource in the old OpenQL to prevent this conflict. HvS presented an algorithm in our common meeting that could be used to generate these resource checks. Why is this comment then still here? This suggests that the implementation accepts that it every now and then fails.

Reply:
The experimentalists in the lab using OpenQL require full control over pulses (and scheduling), and therefore algorithms automating things aren't easily accepted, especially if they are sensed as a black box. Doing things manually - even if that sometimes results in a compiler error - is perceived as being easier (especially at this stage with 17 qubits only), and it allows exploring new ideas, such as a CZ that not only parks some qubit, but also performs 'refocussing', flux assisted measurement, etc.

Reply:
I appreciate that the experimentalists want to be in full control and need support for that. That is the idea behind making passes visible and all functionality transparent: that 'compiler-intelligence' can always be overridden.
But the code generator passes are kind of inevitable: every compiler for transmon needs to run them.
The experimentalists are not the only ones using transmon and its compiler.
For one, there are the QuantumInspire users that like to use a real machine for some reason.
For them, there should be a more fool-proof but also more open, research-encouraging compiler implementation that doesn't run into errors on some circuits.

Reply:
For QI users is may indeed we fruitful to create a separate pass for qubit parking, especially when the number of qubits becomes larger that the current 5 (where conflicts are sort of impossible).
Still I do think it's inevitable that users can construct syntactically sound circuits that still cannot run on some actual hardware, and thus results in an error. The way user/input errors are reported in OpenQL should definitely be improved.

Reply:
Can you make an issue that a compile setup (with new passes or with existing passes with new options) should be created that is suitable for QI users, i.e. that don't crash on a wrong qubit parking setup?
All circuits that users can write and that cannot be run on some actual hardware must get a decent error message; this requires adding passes that check this and then fail with a clear message.

Reply:
done: https://github.com/QuTech-Delft/OpenQL/issues/445